### PR TITLE
Creating backorder

### PR DESCRIPTION
**Pull request by pedrooste**
creates backorder using an online sale. This has been manual tested and confirmed to work. I will work on adding the instore sale side once the stores have been setup / tomororw.

### CODE REVIEW

File: src/main/java/com/mercuryCyclists/Sales/entity/OnlineSale.java [hunk 1]
```diff
@@ -1,20 +1,17 @@
...
 import javax.persistence.Entity;
 import javax.persistence.Table;
 
-
 /**
  * OnlineSale Entity / DAO
  */
 @AllArgsConstructor
 @NoArgsConstructor
+@ToString
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Sales/entity/OnlineSale.java:14]: not needed, will rm later

File: src/main/java/com/mercuryCyclists/Sales/controller/OnlineSaleController.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+@RequestMapping(path = "api/v1/online/sale")
+public class OnlineSaleController {
+
+    private final OnlineSaleService onlineSaleService;
+
+    @Autowired
+    public OnlineSaleController(OnlineSaleService onlineSaleService) {
+        this.onlineSaleService = onlineSaleService;
+    }
+
```
Comment (Jack-Is-2048) [inline src/main/java/com/mercuryCyclists/Sales/controller/OnlineSaleController.java:26]: Add comment explaining method

**Review by Jack-Is-2048 — DISMISSED** [review]
Par the comment, everything else looks good

**Review by Jack-Is-2048 — DISMISSED** [review]
@pedrooste 👍 This PR looks great - it's ready to merge! :shipit:  🚀

**Review by Jack-Is-2048 — APPROVED** [review]
Looks good

**Pull request merged**